SocketExtensions/SocketAwaitable.cs

```diff
@@ -60,10 +61,18 @@ public void OnCompleted(Action continuation)
             }
         }
 
-        public void GetResult()
+        public void GetResult() 
         {
-            if (m_eventArgs.SocketError != SocketError.Success)
-                throw new SocketException((int)m_eventArgs.SocketError);
+            try
+            {
+                if (m_eventArgs.SocketError != SocketError.Success)
+                    throw new SocketException((int)m_eventArgs.SocketError);
+            }
+            catch (SocketException)
```

Comment:
this will suppress all errors and as this is essentially a library file to us it is not okay